src/promptflow-devkit/promptflow/_sdk/_service/apis/ui.py

```diff
@@ -152,7 +139,10 @@ def get_set_flow_yaml(flow, experiment, is_get=True):
             if not os.path.exists(flow):
                 raise UserErrorException(f"The flow doesn't exist: {flow}")
         flow_path = flow
-    return Path(flow_path)
+    flow_path = Path(flow_path)
+    parent_dir = flow_path.parent
+    parent_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
Why is it mkdir parent_dir instead of flow_path?

Reply:
flow_path may be a file here when pass experiment yaml filw